### PR TITLE
FUSETOOLS-2968 - Adding progress monitor

### PR DESCRIPTION
**Pull request by bfitzpat**
-- used approach Aurelien applied in his local repo

Signed-off-by: Brian Fitzpatrick <bfitzpat@redhat.com>

# Pull Request Checklist

After this checklist is all checked or PR provides explanations for possible pass-through, please put the label "Ready for review" on the PR.

## General

- [ ] Did you use the Jira Issue number in the commit comments?
- [ ] Did you set meaningful commit comments on each commit?
- [ ] Did you sign off all commits for this PR? (git commit -s -m "jira issue number - your commit comment")

## Functional

- [ ] Did the CI job report a successful build?
- [ ] Is the non-happy path working, too?

## Maintainability

- [ ] Are all Sonar reported issues fixed or can they be ignored?
- [ ] Is there no duplicated code?
- [ ] Are method-/class-/variable-names meaningful?

## Tests

- [ ] Are there unit-tests?
- [ ] Are there integration tests (or at least a jira to tackle these)?
- [ ] Do we need a new UI test?

## Legal

- [ ] Have you used the correct file header copyright comment?
- [ ] Have you used the correct year in the headers of new files?

### CODE REVIEW

**Comment by bfitzpat** [comment]
I'm not completely sure this is the way we want to go. The actual work done by wsdl2rest is VERY quick and there's a pop-up dialog that blocks the wizard dialog when we refresh the project (adding folders/files). I think this is an improvement, but am open to suggestions.

**Comment by apupier** [comment]
> The actual work done by wsdl2rest is VERY quick

which size of wsdl have you used? How time is increasing when the size/complexity of wsdl is increasing?

**Comment by apupier** [comment]
> there's a pop-up dialog that blocks the wizard dialog when we refresh the project (adding folders/files)

yes, it is better to have the progress monitor integrated inside the wizard instead of having another popup

**Comment by bfitzpat** [comment]
At this point I have one wsdl I test with (address.wsdl), which I would say is on the moderate/light side of complexity. I don't have a great library of more complicated WSDLs to try yet. @tsedmik do you happen to have more complex examples anywhere?

As far as the progress monitor, I agree that it's good to have regardless - but it still seems weird when we refresh the project and the act of doing a refresh after we add some dependencies (MavenUtils().updateMavenDependencies(deps, project);). I will see if maybe we can re-use the progress monitor we're using in the wizard's perform finish to get around that...

**Comment by bfitzpat** [comment]
Updated PR to pass the monitor when we update Dependencies and that seems to get around the extra progress dialog.

**Comment by apupier** [comment]
Wsdl2RestWizardIT test are now failing:

java.lang.NullPointerException: null
	at org.fusesource.ide.wsdl2rest.ui.wizard.Wsdl2RestWizard.performFinish(Wsdl2RestWizard.java:107)
	at org.fusesource.ide.wsdl2rest.ui.tests.integration.Wsdl2RestWizardIT.runWsdl2RestWizard(Wsdl2RestWizardIT.java:106)
	at org.fusesource.ide.wsdl2rest.ui.tests.integration.Wsdl2RestWizardIT.testWsdl2RestWizardWithBlueprintNoProjectInCamelPath(Wsdl2RestWizardIT.java:136)

**Comment by apupier** [comment]
proposal fix for the failing tests: https://github.com/bfitzpat/fuseide/pull/36

File: editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/Wsdl2RestWizard.java [hunk 1]
```diff
@@ -71,31 +74,55 @@
...
 
+	/* (non-Javadoc)
+	 * @see org.eclipse.jface.wizard.Wizard#performFinish()
+	 */
 	@Override
 	public boolean performFinish() {
 		try {
-			generate();
-		} catch (Exception e) {
+			getContainer().run(false, false, new IRunnableWithProgress() {
```
Comment (lhein) [inline editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/Wsdl2RestWizard.java]: Sonar will complain about this one.

Comment (lhein) [reply editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/Wsdl2RestWizard.java]: seems I am wrong :D never mind

**Comment by bfitzpat** [comment]
Thanks for the test fix @apupier !

**Comment by tsedmik** [comment]
@bfitzpat regarding more complex examples. We do not have such kind of examples :-(.

**Pull request merged**